perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -198,7 +209,7 @@ public ApplicationMail getMailById(PerunSession sess, Integer id) throws Interna
 			// set
 			if (mails.size() != 1) {
 				log.error("[MAIL MANAGER] Wrong number of mail definitions returned by unique params, expected 1 but was: {}.", mails.size());
-				throw new InternalErrorException("Wrong number of mail definitions returned by unique params, expected 1 but was: "+mails.size());
+				throw new ApplicationMailNotExistsException("Wrong number of mail definitions returned by unique params, expected 1 but was: "+mails.size());
```

Comment:
This is not the case mail not exists. Not Exists case is catched below in handling ``EmptyResultDataAccessException``. This covers cases, when there are more than 1 mail definitions, so it should be consistency error (probably). You can fix above condition to be more specific ``>1``.